packages/search-ui/src/SearchDriver.js

```diff
@@ -219,85 +222,147 @@ export default class SearchDriver {
       ...searchParameters
     };
 
+    // We want to make sure if this method is running, that any calls to this
+    // method that had been previously deferred are cancelled. We don't want to
+    // possibly have the deferred call execute after this immediate call.
+    this.debounceManager.cancelByName("_updateSearchResults");
+
+    // We bail on making state updates if "isLoading" is true, which implies
+    // that there is an active API request that has been made and is pending.
+    //
+    // The net effect of this is that the user interaction effectively becomes
+    // "locked" while API requests are in flight, to avoid any odd race
+    // conditions.
+    //
+    // NOTE: This may be unnecessary now that we have debounce
+    // "_makeSearchRequest".
```

Comment:
Do we want to consider deleting it in that case? Or leave it in just for now / just in case?

Reply:
I don't know exactly what to do with that 🤔

Reply:
Sorry, as in, you're not sure what my comment means? I was mostly responding to the `This may be unnecessary now` sentence, and wondering if we should then delete the line below (the `isLoading` check).

Reply:
That is what I'm unsure of, whether or not to delete that line.

Reply:
Oh hahaha. I mean we could comment it out with a TODO and see what breaks in a month, and if nothing does delete it for reals? :trollface:

Reply:
It probably warrants some thorough investigation. Honestly, it's a little entangled in everything. Which is why I'm hesitant to take it on right now and also hesitant to remove it.

Reply:
Yep, I'm good with that - my comment was mostly tongue in cheek haha.

Reply:
@constancecchen I did some analysis of this code and ended up removing this check.

As far as I understand, we had the check in there so that no two concurrent requests were ever being made. This can cause complexity because...
1) If a later request completes before a former, the WRONG search results will be shown on the page
2) If there are 2 requests in flight and one finishes (but not the other), the "isLoading" flag will incorrectly be set to false.

These concerns have actually been addressed by the introduction of the `RequestSequencer` inside of this class.

After removing it, I've done some testing and everything looks good. Scarily, no tests broke when I removed it.

This greatly simplifies the code, I'm glad you pointed it out.

Reply:
Awesome!

Quick question - can we remove all instances of isLoading from our state now that we don't seem to be using it to return early?

https://github.com/elastic/search-ui/blob/b0c81b8f5fb9144d10c9801b2e95c637f8cdfce7/packages/search-ui/src/SearchDriver.js#L44

https://github.com/elastic/search-ui/blob/b0c81b8f5fb9144d10c9801b2e95c637f8cdfce7/packages/search-ui/src/SearchDriver.js#L280

https://github.com/elastic/search-ui/blob/b0c81b8f5fb9144d10c9801b2e95c637f8cdfce7/packages/search-ui/src/SearchDriver.js#L311

Unless I'm missing something and it's being passed implicitly down to a child that's using it?...

Reply:
EDIT: Ah okay we do seem to be passing it down

https://github.com/elastic/search-ui/blob/00013212e7a19ca2e086aa147c53a228eff61fab/ADVANCED.md#L131

In which case nevermind me :) Will mark this as resolved

Reply:
Awesome. Thanks for explaining that - this looks great!

Reply:
@constancecchen Yeah, "isLoading" is actually still important for users. It's let's them hook into our loading state logic and show a loading spinner, etc. if they need to.